src/js/providers/html5.js

```diff
@@ -244,6 +253,31 @@ function VideoProvider(_playerId, _playerConfig) {
         }
     }
 
+    function _checkForBuffering(position) {
+        const bufferedRange = _videotag.buffered;
+        if (bufferedRange.length <= 0) {
+            return;
+        }
+        let withinBuffer;
+        if (position <= bufferedRange.end(bufferedRange.length - 1)) {
+            for (let i = 0; i < bufferedRange.length; i++) {
+                withinBuffer = position > bufferedRange.start(i) && position < bufferedRange.end(i);
+                if (withinBuffer) {
+                    break;
```

Comment:
If you move `_videotag.removeEventListener('waiting', setBufferingState);` before this loop, we can replace the `withinBuffer` variable with a return statement. Avoiding declaration of variables whenever possible:

```js
_videotag.removeEventListener('waiting', setBufferingStat
if (position <= bufferedRange.end(bufferedRange.length - 1)) {
    for (let i = 0; i < bufferedRange.length; i++) {
        if(position > bufferedRange.start(i) && position < bufferedRange.end(i);
            return;
}}}
_videotag.addEventListener('waiting', setBufferingState);
```

I don't see the point of adding this event listener only when seeking outside the buffer range. I would either always emit a `STATE_LOADING` event when seeking outside the buffer range OR always listen for 'waiting' to trigger `STATE_LOADING` regardless of `_videotag.buffered` state. Putting these two together just makes the solution more complicated and less performant.

Reply:
If you want to add support for Firefox we could also add an additional listener for "progress" but I would prefer we open a bug with mozilla to get the "waiting" event to match other browsers. "progress" could mean that the other parts of the video are still loading not essential to playback of the part we're seeking to, and "stalled" means that the browser data from the server is not forthcoming - but that data could also be not essential to playback.